### PR TITLE
Support null ClaimDetails response

### LINKED ISSUE #364: Hide Claim Details section if claimDetails object is null
### Description

UIB has determined that in order to be consistent with UIO, claimants who don't match the conditional logic that displays the Claim Summary section on the UIO homepage should also not see the Claim Details section on the Claim Tracker page. These would be the claimants who can only access the Claim Tracker via the link in the global header. 

### Acceptance Criteria

- [ ] When the API returns a null claimDetails object, hide the Claim Details section (including the heading and the fields it contains) from the Claim Tracker page

### PR DESCRIPTION
**Pull request by lomky**
Changes:
 - Updates `getScenarioContent` to handle null claimDetails
 - Updates `ClaimSection` component to display/not display Claim Details based on that
 - Updates the ApiStub to support null claimDetails
 - Adds tests
 - Updates storybook

To Test:
  - `yarn test`
  - Open storybook, open the `Page` component & toggle `hasClaimDetails` to false

Resolves #364 

AC:

 - [x] When the API returns a null claimDetails object, hide the Claim Details section (including the heading and the fields it contains) from the Claim Tracker page

### CODE REVIEW

**Review by kalvinwang — APPROVED** [review]
code lgtm, thanks! looks like yarn build is returning a type error. also, when i run yarn test locally on this branch, i get the following errors... sounds like you don't? might be something specific to my environment...

```
 FAIL  tests/components/ClaimSection.test.tsx
  ● Claim Status & Claim Details are provided › shows both sections

    TypeError: _router.useRouter.mockReturnValue is not a function

      12 |       locale: 'en',
      13 |     }
    > 14 |     ;(useRouter as jest.Mock).mockReturnValue(mockRouter)
         |                               ^
      15 |     const scenarioContent: ScenarioContent = getScenarioContent(apiGatewayStub(ScenarioType.Scenario1))
      16 |
      17 |     render(

      at Object.<anonymous> (tests/components/ClaimSection.test.tsx:14:31)

  ● No Claim Details are provided › hides the Claim Details section

    TypeError: _router.useRouter.mockReturnValue is not a function

      32 |       locale: 'en',
      33 |     }
    > 34 |     ;(useRouter as jest.Mock).mockReturnValue(mockRouter)
         |                               ^
      35 |     const scenarioContent: ScenarioContent = getScenarioContent(apiGatewayStub(ScenarioType.Scenario1, true, false))
      36 |
      37 |     render(

      at Object.<anonymous> (tests/components/ClaimSection.test.tsx:34:31)
```

**Comment by lomky** [comment]
>  looks like yarn build is returning a type error. also, when i run yarn test locally on this branch, i get the following errors... sounds like you don't? might be something specific to my environment...
> 
> ```
>  FAIL  tests/components/ClaimSection.test.tsx
>   ● Claim Status & Claim Details are provided › shows both sections
> 
>     TypeError: _router.useRouter.mockReturnValue is not a function

> ```

Oops - I accidentally snipped out the router mock there during a last minute refactor!

File: components/ClaimSection.tsx [hunk 1]
```diff
@@ -13,6 +13,23 @@ export const ClaimSection: React.FC<ClaimSectionProps> = ({
...
+        programType={detailsContent.programType}
+        benefitYear={detailsContent.benefitYear}
+        claimBalance={detailsContent.claimBalance}
+        weeklyBenefitAmount={detailsContent.weeklyBenefitAmount}
+        lastPaymentIssued={detailsContent.lastPaymentIssued}
+        extensionType={detailsContent.extensionType}
+      />
+    )
+  } else {
+    claimDetails = <div className="claim-details claim-section container" />
```
Comment (rocketnova) [inline components/ClaimSection.tsx:31]: Why do we need to have this empty div?

File: tests/pages/__snapshots__/index.test.tsx.snap [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Exemplar react-test-renderer Snapshot test renders homepage unchanged 1`] = `
+exports[`Full page snapshot renders homepage unchanged 1`] = `
```
Comment (rocketnova) [inline tests/pages/__snapshots__/index.test.tsx.snap:3]: Thank you!

File: types/common.tsx [hunk 3]
```diff
@@ -44,6 +44,7 @@ export interface Claim {
   claimDetails?: null | ClaimDetailsResult
   hasPendingWeeks?: null | undefined | boolean
   hasCertificationWeeksAvailable?: null | undefined | boolean
+  hasClaimDetails?: null | undefined | boolean
```
Comment (rocketnova) [inline types/common.tsx:47]: This doesn't appear to be used. Can it be removed?

File: components/ClaimSection.tsx (hunk 1 above)
Comment (lomky) [reply components/ClaimSection.tsx:31]: That supports the website layout retaining its proper spacing - specifically the `claim-section` bit

File: types/common.tsx (hunk 3 above)
Comment (lomky) [reply types/common.tsx:47]: its not used in the actual flow, but it supports the api stub and the storybook. If there was some better way to handle this I'm super interested though!

Comment (lomky) [reply types/common.tsx:47]: removed it in #375

**Pull request merged**